rocAL/include/api/rocal_api_data_loaders.h

```diff
@@ -617,6 +617,58 @@ extern "C" RocalTensor ROCAL_API_CALL rocalRawTFRecordSourceSingleShard(RocalCon
                                                                         const char* record_name_prefix = "",
                                                                         RocalShardingInfo rocal_sharding_info = RocalShardingInfo());
 
+/*! \brief Creates Numpy raw data reader and loader. It allocates the resources and objects required to read raw data stored on the numpy arrays.
+ * \ingroup group_rocal_data_loaders
+ * \param [in] context Rocal context
+ * \param [in] source_path A NULL terminated char string pointing to the location on the disk
+ * \param [in] internal_shard_count Defines the parallelism level by internally sharding the input dataset and load using multiple loader instances. Using shard counts bigger than 1 improves the load performance if compute resources (CPU cores) are available.
+ * \param [in] output_layout the layout to be set for the input tensor
+ * \param [in] files Contains a list of file paths to read the data from.
+ * \param [in] is_output Determines if the user wants the loaded images to be part of the output or not.
+ * \param [in] shuffle Determines if the user wants to shuffle the dataset or not.
+ * \param [in] loop Determines if the user wants to indefinitely loops through images or not.
+ * \param [in] seed Determines the seed used by RNG for shuffling data between shards.
+ * \param [in] rocal_sharding_info The members of RocalShardingInfo determines how the data is distributed among the shards and how the last batch is processed by the pipeline.
+ * \return Reference to the output tensor
+ */
```

Comment:
Rajy had suggested this earlier whichever API we are using we can retain that alone

Reply:
We use this API in the C++ unit tests for numpy reader test case

Reply:
But can we not use the other API in C++?
Do we need to have both please check once with Shobana if needed